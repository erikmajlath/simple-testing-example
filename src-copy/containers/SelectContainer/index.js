import React from 'react'
import Select, { OPTIONS } from 'components/Select'

export default class SimpleSelect extends React.Component {
  state = {
    value: ['sf'],
  }

  onSelectChange = value => this.setState({ value })

  render() {
    return (
      <Select
        options={OPTIONS}
        value={this.state.value}
        onChange={this.onSelectChange}
      />
    )
  }
}
