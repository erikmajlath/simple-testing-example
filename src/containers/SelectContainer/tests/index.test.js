import React from 'react'
import { shallow } from 'enzyme'
import SelectContainer from '../index'

describe('SelectContainer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<SelectContainer />)
  })

  it('passes state value to select', () => {
    wrapper.setState({ value: ['sf', 'ny'] })
    
    expect(wrapper).toMatchSnapshot()
  })
})
