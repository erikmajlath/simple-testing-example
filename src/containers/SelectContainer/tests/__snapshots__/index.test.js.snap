// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectContainer passes state value to select 1`] = `
<Component
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "San Francisco",
        "value": "sf",
      },
      Object {
        "label": "Los Angeles",
        "value": "la",
      },
      Object {
        "label": "New York City",
        "value": "ny",
      },
    ]
  }
  value={
    Array [
      "sf",
      "ny",
    ]
  }
/>
`;
